jkube-kit/profile/src/main/java/org/eclipse/jkube/kit/profile/ProfileUtil.java

```diff
@@ -100,16 +101,19 @@ private static Profile inheritFromParentProfile(Profile aProfile, File resourceD
      * @param resourceDirs resource directory where to lookup the profile (in addition to a classpath lookup)
      * @return the merged configuration which can be empty if no profile is given
      * @param config the provided configuration
-     * @throws IOException
      */
     public static ProcessorConfig blendProfileWithConfiguration(ProcessorConfigurationExtractor configExtractor,
                                                                 String profile,
                                                                 List<File> resourceDirs,
-                                                                ProcessorConfig config) throws IOException {
-        // Get specified profile or the default profile
-        ProcessorConfig profileConfig = extractProcesssorConfiguration(configExtractor, profile, resourceDirs);
+                                                                ProcessorConfig config) {
+        try {
+            // Get specified profile or the default profile
+            ProcessorConfig profileConfig = extractProcesssorConfiguration(configExtractor, profile, resourceDirs);
 
-        return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+            return ProcessorConfig.mergeProcessorConfigs(config, profileConfig);
+        } catch (IOException ioException) {
+            throw JKubeException.launderThrowable("Error in extracting processor configuration", ioException);
+        }
     }
```

Comment:
If you carefully check the method, the IOExcpetion is only thrown at the `public static Profile findProfile(String profileArg, File... resourceDirs)` method.

Maybe this is the method where we want to throw the JKubeException instead.